dataframely/columns/array.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) QuantCo 2025-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+from collections.abc import Callable, Sequence
+from typing import Any, Literal, cast
+
+import polars as pl
+
+from dataframely._compat import pa, sa, sa_TypeEngine
+from dataframely.random import Generator
+
+from ._base import Column
+from .struct import Struct
+
+
+class Array(Column):
+    """A fixed-shape array column."""
+
+    def __init__(
+        self,
+        inner: Column,
+        shape: int | tuple[int, ...],
+        *,
+        nullable: bool = True,
+        # polars doesn't yet support grouping by arrays,
+        # see https://github.com/pola-rs/polars/blob/810df456/crates/polars-core/src/frame/group_by/into_groups.rs#L365
```

Comment:
Ideally, we'd be linking an issue here so that we can trace more easily when it might be resolved. If none exists, could you create one? :)

Reply:
I just opened https://github.com/pola-rs/polars/issues/22574 and it seems it will be implemented soon!

Reply:
Damn, that was fast :smile